### DISCUSSION TITLE
Support for new shard of oss.sonatype - s01.oss.sonatype.org

### DISCUSSION THREAD

**Discussion by ghostbuster91** [com-lihaoyi/mill#1225] [Q&A] [answered]
Hi,

According to https://central.sonatype.org/articles/2021/Feb/23/new-users-on-s01osssonatypeorg/ new users are given access to s01.oss.sonatype.org and have to perform publishing through this server. This issue emerged also on sbt-sonatype repository: https://github.com/xerial/sbt-sonatype/issues/214 

I think the solution presented by @steinybot  in https://github.com/xerial/sbt-sonatype/pull/215 is quite good.

I will be happy to contribute that change, but before jumping into the code I would like to hear your opinion.

Cheers,
Kasper

**Comment by lefou** [comment, marked as answer]
The `mill.scalalib.publish.SonatypePublisher` already supports parametrized repository urls. The default urls are defined in `mill.scalalib.PushlishModule.sonatypeUri` and `mill.scalalib.PushlishModule.sonatypeSnapshotUri`, and can be overridden  in your module.

**Comment by ghostbuster91** [comment]
Hi, thanks for the tip. It indeed worked. For any future readers here is what I did:
```
trait MyPublishModule extends PublishModule {
  def sonatypeUri: String = "https://s01.oss.sonatype.org/service/local"

  def sonatypeSnapshotUri: String =
    "https://s01.oss.sonatype.org/content/repositories/snapshots"
}
```


Unfortunately, this configuration isn't being picked up when using publishAll syntax and has to be duplicated:
```
./mill -i mill.scalalib.PublishModule/publishAll --publishArtifacts __.publishArtifacts --sonatypeUri "https://s01.oss.sonatype.org/service/local" 
```

Is there any way to workaround that?

**Reply by lefou** [reply]
> Unfortunately, this configuration isn't being picked up when using publishAll syntax and has to be duplicated:
> 
> ```
> ./mill -i mill.scalalib.PublishModule/publishAll --publishArtifacts __.publishArtifacts --sonatypeUri "https://s01.oss.sonatype.org/service/local" 
> ```
> 
> Is there any way to workaround that?

No, this is currently hardcoded, but you could create your own publishAll command, delegating the work to the `PublishModule` but using your preferred default values.

**Reply by ghostbuster91** [reply]
Ok, I will look into that, thanks again :)
